File: core/src/main/java/com/mcreater/amclcore/model/oauth/XBLTokenResponseModel.java
package com.mcreater.amclcore.model.oauth;

import com.mcreater.amclcore.annotations.OAuthLoginModel;
import lombok.Builder;
import lombok.Data;

@Builder
@Data
@OAuthLoginModel
public class XBLTokenResponseModel {
    private XBLTokenResponsePropertiesModel Properties;
    private String RelyingParty;
    private String TokenType;

    @Builder
    @Data
    @OAuthLoginModel
    public static class XBLTokenResponsePropertiesModel {
        private String AuthMethod;
        private String SiteName;
        private String RpsTicket;
    }
}

File: core/src/main/java/com/mcreater/amclcore/model/oauth/XBLTokenResponseModel.kt
package com.mcreater.amclcore.model.oauth

import com.mcreater.amclcore.annotations.OAuthLoginModel

@OAuthLoginModel
data class XBLTokenResponseModel(
    private var Properties: XBLTokenResponsePropertiesModel? = null,
    var RelyingParty: String? = null,
    var TokenType: String? = null
) {
    @OAuthLoginModel
    data class XBLTokenResponsePropertiesModel(
        var AuthMethod: String? = null,
        var SiteName: String? = null,
        var RpsTicket: String? = null
    )
}